### PR TITLE
[Caching] Add new storage type elements that create child BlueprintViews

### PR DESCRIPTION
**Pull request by kdaker**
Adds new init for `ElementContent` that takes and measures a child element. It uses a new storage type that only caches the measurement for the child element but does not return any layout nodes.

### CODE REVIEW

File: BlueprintUI/Sources/Element/ElementContent.swift [hunk 1]
```diff
@@ -91,6 +91,53 @@ extension ElementContent {
...
+    /// Creates a new `ElementContent` which uses the provided element to measure its
+    /// size, but does not place the element as a child in the final, laid out hierarchy.
+    ///
+    /// This is useful if you are placing the element in a nested `BlueprintView`, for example (eg
+    /// to create a stateful element) and just need this element to be correctly sized.
+    public init(byMeasuring element: Element) {
+        storage = MeasureNestedElementStorage(element: element)
+    }
+
+    struct MeasureNestedElementStorage: ContentStorage {
```
Comment (kyleve) [inline BlueprintUI/Sources/Element/ElementContent.swift:105]: Nit: private?

File: BlueprintUI/Sources/Element/ElementState.swift [hunk 2]
```diff
@@ -235,12 +251,14 @@ final class ElementState {
         identifier: ElementIdentifier,
         element: Element,
         signpostRef: AnyObject,
-        name: String
+        name: String,
+        kind: Kind = .regular
```
Comment (kyleve) [inline BlueprintUI/Sources/Element/ElementState.swift:255]: IMO given this is an internal-only method, I think requiring folks to pass a value is best, vs them forgetting and getting it wrong

**Review by kyleve — APPROVED** [review]

File: BlueprintUI/Sources/Element/ElementContent.swift [hunk 3]
```diff
@@ -111,6 +111,65 @@ extension ElementContent {
...
+
+        func performLayout(
+            in size: CGSize,
+            with environment: Environment,
+            state: ElementState
+        ) -> [LayoutResultNode] {
+            []
+        }
+
+        func forEachElement(
```
Comment (n8chur) [inline BlueprintUI/Sources/Element/ElementContent.swift:159]: How is this used?

Comment (kyleve) [reply BlueprintUI/Sources/Element/ElementContent.swift:159]: We use this to enumerate the element tree, to make sure that the newest element makes it into the live state; even if we return a cached layout, eg: https://github.com/square/Blueprint/pull/404/files#diff-87a9ff4712c4ac0a865f6587fa32cecfb76335e0767e569de804727956baffc4L489

**Pull request merged**